### ISSUE TITLE
Cannot find library 'Scheduler.dll'

### ISSUE THREAD

**Issue by alexwoolfson** [jvmahon/Homebridge-HomeSeer4#35]
There's actually one more issue I have been experiencing since I set up Homebridge-HomeSeer4.

I have a script I run in HS3 that slowly raises the brightness of a floor lamp when I wake up in the morning. Since I installed Homebridge and this plugin a couple days ago, it's been broken.

Checking the logs, I found this

```
Jan-05 10:52:03 AM	 	Event	Running script in background: C:/Program Files (x86)/HomeSeer HS3/scripts/WakeUpLightRamp.vb("Main","57")
Jan-05 10:52:03 AM	 	Event	Event Trigger "Virtual General Status Wake Alex Up - Increase Bedroom Light 1% every 6 seconds"
Jan-05 10:51:57 AM	 	Error	Compiling script C:\Program Files (x86)\HomeSeer HS3\scripts\WakeUpLightRamp.vb: compiler initialization failed unexpectedly: The system cannot find the file specified.
Jan-05 10:51:57 AM	 	Error	Compiling script C:\Program Files (x86)\HomeSeer HS3\scripts\WakeUpLightRamp.vb: could not find library 'Scheduler.dll'
```

When I set things up, I followed these instructions to get things to automatically launch:

https://github.com/jvmahon/Homebridge-HomeSeer4/wiki/Automatically-Start-HomeSeer-and-Homebridge-at-Windows-Startup

And per your recommendation, I followed these steps to make sure the priority was set to Normal for both HS3 and Homebridge (I chose priority "4"):

https://bdbits.wordpress.com/2010/04/29/setting-a-scheduled-task-process-priority/

Trying to troubleshoot this issue, I found these two Homeseer threads which make me think that how HS3 is now starting up could be the source of the problem:

https://forums.homeseer.com/forum/homeseer-products-services/system-software-controllers/hs3-hs3pro-software/hs3-hs3pro-discussion/103478-script-can-t-find-scheduler-dll

https://forums.homeseer.com/forum/homeseer-products-services/system-software-controllers/hs3-hs3pro-software/hs3-hs3pro-discussion/78555-startup-error-startup-vb-could-not-find-library-scheduler-dll-solution

I realize that this is a Windows things and not a result of your awesome plugin. But I'm wondering since you've also set up your system this way, if you've run into this issue before and if you know of a solution. 

If so, please let me know.

Thank you again for the awesome plugin. :)

**Comment by jvmahon** [comment]
Post a copy of  the script so I can try it on my system and then I can get a better idea of what’s wrong.

**Comment by alexwoolfson** [comment]
Thank you for trying to help me with this. I really appreciate it. :)

This is the script:

```
Public Sub Main(ByVal _deviceRef As Object)
        Dim deviceRef As Integer = Integer.Parse(_deviceRef)
        Dim deviceValue As Integer = hs.DeviceValue(deviceRef)

        'Exit if the device is already at maximum, i.e. "On"
        If hs.CAPIGetStatus(deviceRef).Status = "On" Then Exit Sub

        'Find the correct CAPI control to execute. I really miss Linq!!!
        For Each cc As HomeSeerAPI.CAPIControl In hs.CAPIGetControl(deviceRef)
            If cc.ControlValue = (deviceValue + 1) Then

                'Set the correct dim level
                hs.CAPIControlHandler(cc)

                'We're done, so let's exit!
                Exit Sub
            End If
        Next
    End Sub
```

This is the event:

<img width="953" alt="2020-01-05_11-10-09" src="https://user-images.githubusercontent.com/1188997/71784766-069c2c00-2fac-11ea-9d9f-d16741ced230.png">

Please let me know if you need anything else. It's been years since I set this up and it was something I basically copied from a HomeSeer forum thread.

Thanks!

Alex

**Comment by jvmahon** [comment]
I vaguely remember having that problem at some point and I think the resolution was what was suggested in your second link (https://forums.homeseer.com/forum/homeseer-products-services/system-software-controllers/hs3-hs3pro-software/hs3-hs3pro-discussion/78555-startup-error-startup-vb-could-not-find-library-scheduler-dll-solution?p=805249#post805249)

**Comment by jvmahon** [comment]
I've also seen a recommendation to un-install and re-install HomeSeer.

**Comment by alexwoolfson** [comment]
Thank you for getting back to me. I really appreciate the help. 

I'm looking in the Task Scheduler for what I set up to automatically launch HomeSeer based on the wiki instructions and I don't see a "Start in" field. (There is a "Location:" field, though.)

If you think you've done this before, would it be possible for you to share a screenshot with me of what you added to your scheduled task to launch HomeSeer to avoid this problem?

Also, I've never un-installed and re-installed HomeSeer. Would doing that mean I'd need to reset everything up again?

Thank you again for your help,

Alex

**Comment by jvmahon** [comment]
It is called "Start in (optional):"

Look in Section D of the Wiki here:  https://github.com/jvmahon/Homebridge-HomeSeer4/wiki/Automatically-Start-HomeSeer-and-Homebridge-at-Windows-Startup.

Its shown in the 3rd image.  Note that if those images are for when you use the "Wizard" to initially set up a task for startup.  If the task is already set up, then go to the task in "Task Scheduler" and then go to the "Actions" pane for the task. You should see a "Start a program action" 

![image](https://user-images.githubusercontent.com/15061942/72003368-37cc6480-3217-11ea-93a4-271c2ee2a18e.png)

**Comment by jvmahon** [comment]
As for updating HomeSeer, you don't need to fully uninstall it. Just download and re-install the latest update from the HomeSeer web page. https://homeseer.com/current-downloads/

You should not need to set up everything again. But just in case you have a problem, first backup your configuration.

You can do that backup from the HomeSeer web page go to [Tools] -> [Setup], select the "General" tab on the web page that appears, then select "Backup Config".

**Comment by alexwoolfson** [comment]
Yay! You're a rock star. Thank you very much for the hand-holding through the process.

Following your instructions, I was able to find the Start In option by editing my already-made task.

I added "C:\Program Files (x86)\HomeSeer HS3 in that Start In location, restarted my NUC, and now the script works smooth as silk. Everything else works great as well. So no need to uninstall/reinstall HS3.

Thank you again for the above-and-beyond assistance. :)

Alex
